R/loading_button.R

```diff
@@ -0,0 +1,92 @@
+#' loading_button
+#'
+#' Button that becomes disabled until reset w/ `reset_loading_button`
+#'
+#' @param id the flipbox id
+#' @param label the button text (label)
+#' @param style style for button (pre-loading); character string w/ CSS styling format: "color: black; background-color: red;"
+#' @param loading_text text to show after button is clicked (e.g. during loading)
+#' @param loading_style style for button (while loading); character string w/ CSS styling format: "color: black; background-color: red;"
+#'
+#' @export
+#'
+#' @importFrom htmltools tags
+#' @family flip_box functions
+
+loading_button <- function(id, 
+                           label,
+                           style = "color: #FFFFFF; background-color: #337AB7; border-color: #2E6DA4;",
+                           loading_text = "Loading...",
+                           loading_style = "color: #FFFFFF; background-color: grey; border-color: grey;"
+) {
+  
+  htmltools::tagList(
+    tags$button(
+      id = paste0("loading_button-", id),
+      class = "btn style",
```

Comment:
I would replace the value for class here with an additional `class` argument to `loading_button()`

Reply:
f6448b6003a856febc94fe254872f15ec9e79b1f